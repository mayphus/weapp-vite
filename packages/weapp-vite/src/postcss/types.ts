import type { MpPlatform } from '@/types'

export interface CssPostProcessOptions {
  platform: MpPlatform
}
