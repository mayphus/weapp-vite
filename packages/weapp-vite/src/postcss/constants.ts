export const cssAtRulePrefix = 'wv'

export const IFDEF = '#ifdef'

export const IFNDEF = '#ifndef'

export const ENDIF = '#endif'
