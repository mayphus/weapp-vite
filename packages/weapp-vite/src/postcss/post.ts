import type { PluginCreator } from 'postcss'

export const postCreator: PluginCreator<unknown> = () => {
  return {
    postcssPlugin: 'postcss-weapp-vite-plugin-post',
    OnceExit(root) {
      root.walkAtRules(/^weapp-vite/, (rule) => {
        if (rule.name === 'weapp-vite-keep-import') {
          rule.name = 'import'
        }
      })
    },
  }
}

postCreator.postcss = true
