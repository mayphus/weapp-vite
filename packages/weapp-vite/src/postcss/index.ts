import postcss from 'postcss'
import { postCreator } from './post'

export function cssPostProcess(css: string) {
  return postcss([postCreator()]).process(css).async()
}
