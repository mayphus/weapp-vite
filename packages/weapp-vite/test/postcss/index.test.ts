import { postCreator } from '@/postcss/post'
import postcss from 'postcss'

describe('postcss', () => {
  it('@weapp-vite-keep-import', async () => {
    const { css } = await postcss([postCreator()]).process('@weapp-vite-keep-import', { from: undefined })
    expect(css).toMatchSnapshot()
  })

  it('@weapp-vite-keep-import case 0', async () => {
    const { css } = await postcss([postCreator()]).process('@weapp-vite-keep-import \'xxx\'', { from: undefined })
    expect(css).toMatchSnapshot()
  })
})
