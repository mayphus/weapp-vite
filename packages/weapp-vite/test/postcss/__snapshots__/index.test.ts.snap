// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`postcss > @weapp-vite-keep-import 1`] = `"@import"`;

exports[`postcss > @weapp-vite-keep-import case 0 1`] = `"@import 'xxx'"`;
